docs/pipelines.md

```diff
@@ -116,75 +117,87 @@ cohortextractor run dummy generate_cohorts expectations
 This will generate the `input.csv` file as explained in the [cohortextractor](cohortextractor.md) section.
 
 To run the second action you can use:
-```
+```sh
 cohortextractor run dummy run_model expectations
 ```
 As this action depends on the `generate_cohorts` action, it will cause `generate_cohorts` to be run first, followed by `run_model`.
-It will create the two files
+It will create the two files.
 
 To run all actions, you can use a special `run_all` action which is created for you (no need to define it in your `project.yaml`):
-```
+```sh
 cohortextractor run dummy run_all expectations
 ```
 
 You will also find a new folder, `metadata/`, which contains logging information about your run. If any of your actions fail, you may find clues here as to why.
 
+Each action is run in its own isolated environment in a temporary working directory. This means that all the necessary libraries and data must be imported within the script for each action &mdash; For R users, this essentially means that the R session is restarted for each action. 
+
+### Dependencies
+
+If one or more dependencies of an action have not been run (i.e., their outputs do not exist) then these dependency actions will be run first. If a dependency has changed but has not been run (so the outputs are not up-to-date with the changes), then the dependent actions will be run on the old outputs. To force the dependencies to be run, use the `--force-run-dependencies` option.
+
+For example `cohortextractor run run_model --force-dependencies` will ensure that both the `run_model` and `generate_cohorts` actions are run, even if `input.csv` already exists.
```

Comment:
This command isn't quite right. If you're doing it through the `cohortextractor` then you need the fairly hideous command:
```
cohortextractor run dummy run_model expectations --force-run --force-run-dependencies
```

The new command will accept:
```
opensafely run run_model --force-run-dependencies
```

Or the even shorter:
```
opensafely run run_model -f
```

Reply:
I've corrected the `cohortextractor` command, pending a re-write for `opensafely`